Basic usage of custom Extra fails to compile

Just changing `X = Extra` to `usize` makes compilation fail.

# Non-working variant #
```rust
use glyph_brush::{
    ab_glyph::FontArc, Extra, GlyphBrush, GlyphBrushBuilder, OwnedSection, OwnedText,
};
let dejavu = FontArc::try_from_slice(include_bytes!("../DejaVuSans.ttf")).unwrap();
let mut brush: GlyphBrush<(), _> = GlyphBrushBuilder::using_font(dejavu).build();
brush.queue(&OwnedSection::default().add_text(
    OwnedText::new("test").with_scale(30.0).with_extra(0usize),
));
```

# Working variant #
```rust
use glyph_brush::{
    ab_glyph::FontArc, Extra, GlyphBrush, GlyphBrushBuilder, OwnedSection, OwnedText,
};
let dejavu = FontArc::try_from_slice(include_bytes!("../DejaVuSans.ttf")).unwrap();
let mut brush: GlyphBrush<(), _> = GlyphBrushBuilder::using_font(dejavu).build();
brush.queue(&OwnedSection::default().add_text(
    OwnedText::new("test").with_scale(30.0).with_extra(Extra {
        z: 0.0,
        color: [0.0; 4],
    }),
));
```

# Compiler error #
```
error[E0277]: the trait bound `Cow<'_, glyph_brush::Section<'_, _>>: From<&OwnedSection<usize>>` is not satisfied
   --> <crate>/src/main.rs:132:13
    |
132 |   brush.queue(&OwnedSection::default().add_text(
    |  _______-----_^
    | |       |
    | |       required by a bound introduced by this call
133 | |     OwnedText::new("test").with_scale(30.0).with_extra(0usize),
134 | | ));
    | |_^ the trait `From<&OwnedSection<usize>>` is not implemented for `Cow<'_, glyph_brush::Section<'_, _>>`, which is required by `&OwnedSection<usize>: Into<Cow<'_, glyph_brush::Section<'_, _>>>`
    |
    = help: the following other types implement trait `From<T>`:
              `Cow<'a, CStr>` implements `From<&'a CStr>`
              `Cow<'a, CStr>` implements `From<&'a CString>`
              `Cow<'a, CStr>` implements `From<CString>`
              `Cow<'a, OsStr>` implements `From<&'a OsStr>`
              `Cow<'a, OsStr>` implements `From<&'a OsString>`
              `Cow<'a, OsStr>` implements `From<OsString>`
              `Cow<'a, Path>` implements `From<&'a Path>`
              `Cow<'a, Path>` implements `From<&'a PathBuf>`
            and 23 others
    = note: required for `&OwnedSection<usize>` to implement `Into<Cow<'_, glyph_brush::Section<'_, _>>>`
note: required by a bound in `GlyphBrush::<V, X, F, H>::queue`
   --> /home/<user>/.cargo/registry/src/index.crates.io-6f17d22bba15001f/glyph_brush-0.7.9/src/glyph_brush.rs:217:12
    |
214 |     pub fn queue<'a, S>(&mut self, section: S)
    |            ----- required by a bound in this associated function
...
217 |         S: Into<Cow<'a, Section<'a, X>>>,
    |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `GlyphBrush::<V, X, F, H>::queue`

```

--- a/glyph-brush/CHANGELOG.md
+++ b/glyph-brush/CHANGELOG.md
@@ -1,3 +1,6 @@
+# Unreleased (0.7.10)
+* Fix conversions from OwnedSection -> Section with non-default extra type.
+
 # 0.7.9
 * Add `GlyphCruncher::glyphs` doc example of using `section_index` & `byte_index`.
 * Update _rustc-hash_ to `2`.

--- a/glyph-brush/src/section/owned.rs
+++ b/glyph-brush/src/section/owned.rs
@@ -73,14 +73,26 @@ impl<X: Clone> OwnedSection<X> {
     }
 }
 
-impl<'a> From<&'a OwnedSection> for Section<'a> {
-    fn from(owned: &'a OwnedSection) -> Self {
+impl<'a, X: Clone> From<&'a OwnedSection<X>> for Section<'a, X> {
+    /// ```
+    /// use glyph_brush::{OwnedSection, OwnedText, Section};
+    ///
+    /// let section = OwnedSection::default().add_text(OwnedText::new("foo").with_extra(1usize));
+    /// let cow: Section<'_, usize> = (&section).into();
+    /// ```
+    fn from(owned: &'a OwnedSection<X>) -> Self {
         owned.to_borrowed()
     }
 }
 
-impl<'a> From<&'a OwnedSection> for Cow<'a, Section<'a>> {
-    fn from(owned: &'a OwnedSection) -> Self {
+impl<'a, X: Clone> From<&'a OwnedSection<X>> for Cow<'a, Section<'a, X>> {
+    /// ```
+    /// use glyph_brush::{OwnedSection, OwnedText, Section};
+    ///
+    /// let section = OwnedSection::default().add_text(OwnedText::new("foo").with_extra(1usize));
+    /// let cow: std::borrow::Cow<'_, Section<'_, usize>> = (&section).into();
+    /// ```
+    fn from(owned: &'a OwnedSection<X>) -> Self {
         Cow::Owned(owned.to_borrowed())
     }
 }

